src/qt/optionsdialog.cpp

```diff
@@ -406,9 +406,8 @@ void OptionsDialog::updateProxyValidationState()
 
 void OptionsDialog::updateDefaultProxyNets()
 {
-    CNetAddr ui_proxy_netaddr;
-    LookupHost(ui->proxyIp->text().toStdString(), ui_proxy_netaddr, /*fAllowLookup=*/false);
-    const CService ui_proxy{ui_proxy_netaddr, ui->proxyPort->text().toUShort()};
+    const std::optional<CNetAddr> ui_proxy_netaddr{LookupHost(ui->proxyIp->text().toStdString(), /*fAllowLookup=*/false)};
+    const CService ui_proxy{ui_proxy_netaddr.value(), ui->proxyPort->text().toUShort()};
```

Comment:
That looks dangerous -- generally I think we shouldn't ever call `.value()` on a std::optional object without checking for `.has_value()` first (similar as for and pointer dereferences), potentially risking a crash. In this case the GUI flow seems to ensure that the IP passed here is valid at an earlier stage (via the `ProxyAddressValidator`) so `LookupHost` _should_ always return a value, but I'd prefer to be rather safe than sorry.

Reply:
You're right. Considering it comes from GUI, it's better to check whether it has value. Addressed it!